.github/workflows/sonar-backend.yaml

```diff
@@ -0,0 +1,64 @@
+name: SonarCloud Scan for backend
+on:
+  push:
+    branches:
+      - "main"
+  pull_request_target:
+    types: [ "opened", "edited", "reopened", "synchronize" ]
+    paths:
+      - "odd-platform-api/**"
+      - "odd-platform-api-contract/**"
+      - "build.gradle"
+
+jobs:
+  images:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v2
+        with:
+          fetch-depth: 1
+      - uses: unfor19/install-aws-cli-action@v1
+
+      - uses: actions/setup-node@v2
+        with:
+          node-version: '14'
+
+      - name: Set up JDK 17
+        uses: actions/setup-java@v2
+        with:
+          java-version: '17'
+          distribution: 'temurin'
+
+      - name: Cache SonarCloud packages
+        uses: actions/cache@v1
+        with:
+          path: ~/.sonar/cache
+          key: ${{ runner.os }}-sonar
+          restore-keys: ${{ runner.os }}-sonar
+
+      - name: Gradle cache
+        uses: actions/cache@v2
+        with:
+          path: |
+            ~/.gradle/caches
+            ~/.gradle/wrapper
+          key: ${{ runner.os }}-shared-gradle-${{ hashFiles('**/*.gradle*', '**/gradle-wrapper.properties') }}
+          restore-keys: |
+            ${{ runner.os }}-gradle-
+
+      - name: Node cache
+        uses: actions/cache@v2
+        with:
+          path: odd-platform-ui/.npm
+          key: ${{ runner.os }}-shared-node-${{ hashFiles('**/package-lock.json') }}
+          restore-keys: |
+            ${{ runner.os }}-node-
+
+      - name: Build with Gradle
+        env:
+          GENERATE_SOURCEMAP: false
+          CI: false
+          JAVA_OPTS: "-Xms2048m -Xmx2048m"
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }} # Needed to get PR information
+          SONAR_TOKEN: ${{ secrets.SONAR_TOKEN }}
+        run: ./gradlew -PskipUIBundle=true -Pversion=ci-${GITHUB_SHA::6} -Penv=prod -x compileTestJava clean :odd-platform-api:sonarqube --info
```

Comment:
is `-Pversion` flag necessary here?

Reply:
Guess not. I've just copy-pasted from another workflow